### PR TITLE
Generate linux syscall constants

### PR DESCRIPTION
**Pull request by mahkoh**
Tl;dr: Many syscall constants were missing so I've written a small script that generates the constants from kernel sources.

As a first step I've reduced the duplication between glibc and musl by creating a new sibling module which is supposed to contain kernel structures and constants. There are many of those that could be moved out of the libc modules but for now I've only moved the syscall constants.

This violates the bottom-up design of the library but this design is not suitable for a system with multiple libcs because a significant part of their surface will be identical reexports of kernel structures and constants.

### CODE REVIEW

**Comment by JohnTitor** [comment]
Hey, thanks for the PR! But this is a major change to our process, so I think we need to discuss it before reviewing.

**Comment by mahkoh** [comment]
Just to be clear: The syscalls are generated manually and not during the build. So the process stays the same unless you want to add new syscalls which you should then do by using the script.

**Comment by mahkoh** [comment]
@JohnTitor What's the status of this?

**Comment by JohnTitor** [comment]
That's great that we could cover many syscalls here but I'm reluctant to this because this will affect too many targets at once.

**Comment by mahkoh** [comment]
Adding new linux syscall numbers affects all linux targets? This is your objection?

**Comment by mahkoh** [comment]
@JohnTitor Can I assume that this PR will stay in limbo indefinitely? You've said that it needs to be "discussed" but I've seen no such discussion nor any concerns raised beyond "it affects all linux targets". Please close the PR if you have no interest in the changes.

**Comment by JohnTitor** [comment]
Sorry for my absence! So, as I said this would affect the targets broadly, and I concerned that this may introduce some inconsistencies. Since the current workflow is fine to maintain, I'm going to close this.
If you still want to add such a change, you could open an issue to make a decision at first.
Thanks for contributing anyway!

**Comment by kpcyrd** [comment]
In case this is getting revived, the script I'm using to generate the syscall list for the syscallz crate is located here: https://github.com/kpcyrd/syscallz-rs/blob/main/gen.py

This is lightly tested on the major debian architectures: https://buildd.debian.org/status/package.php?p=rust-syscallz
